File: TP1/Exo 1.8/Python/calcul.py
# calculs.py

def calculer(a, b, op):
    try:
        # Utilisation de if-elif pour déterminer l'opération à effectuer
        if op == '+':
            return a + b
        elif op == '-':
            return a - b
        elif op == '*':
            return a * b
        elif op == '/':
            # Gestion de la division par zéro
            if b == 0:
                raise ValueError("Erreur : Division par zéro")
            return a / b
        elif op == '%':
            if b == 0:
                raise ValueError("Erreur : Division par zéro")
            return a % b
        else:
            # Gestion de l'opérateur non reconnu
            raise ValueError(f"Erreur : Opérateur non reconnu '{op}'")
    except ValueError as e:
        return str(e)

def main():
    # Lecture des entrées utilisateur
    try:
        a = float(input("Entrez le premier nombre : "))
        b = float(input("Entrez le second nombre : "))
        op = input("Entrez l'opérateur (+, -, *, /, %) : ")

        # Calcul et affichage du résultat
        resultat = calculer(a, b, op)
        print(f"Le résultat de {a} {op} {b} est : {resultat}")

    except ValueError as e:
        print(f"Erreur de saisie : {e}")

if __name__ == "__main__":
    main()

File: TP1/Exo 1.8/C/calcul.c
#include <stdio.h>

// Fonction principale
int main() {
    int num1, num2, result;
    char op;

    // Lecture des entrées utilisateur
    printf("Entrez le premier nombre : ");
    scanf("%d", &num1);
    printf("Entrez le second nombre : ");
    scanf("%d", &num2);
    printf("Entrez l'opérateur (+, -, *, /, %%, &, |, ~) : ");
    scanf(" %c", &op);

    // Utilisation de switch pour déterminer l'opération
    switch (op) {
        case '+':
            result = num1 + num2;
            printf("Résultat de %d %c %d = %d\n", num1, op, num2, result);
            break;
        case '-':
            result = num1 - num2;
            printf("Résultat de %d %c %d = %d\n", num1, op, num2, result);
            break;
        case '*':
            result = num1 * num2;
            printf("Résultat de %d %c %d = %d\n", num1, op, num2, result);
            break;
        case '/':
            if (num2 == 0) {
                printf("Erreur : Division par zéro\n");
            } else {
                result = num1 / num2;
                printf("Résultat de %d %c %d = %d\n", num1, op, num2, result);
            }
            break;
        case '%':
            if (num2 == 0) {
                printf("Erreur : Division par zéro\n");
            } else {
                result = num1 % num2;
                printf("Résultat de %d %c %d = %d\n", num1, op, num2, result);
            }
            break;
        case '&':
            result = num1 & num2;
            printf("Résultat de %d %c %d = %d\n", num1, op, num2, result);
            break;
        case '|':
            result = num1 | num2;
            printf("Résultat de %d %c %d = %d\n", num1, op, num2, result);
            break;
        case '~':
            // Opération NOT sur num1 seulement
            result = ~num1;
            printf("Résultat de ~%d = %d\n", num1, result);
            break;
        default:
            printf("Erreur : Opérateur non reconnu '%c'\n", op);
            break;
    }

    return 0;
}
